src/home_robot/home_robot/mapping/voxel.py

```diff
@@ -424,6 +463,27 @@ def get_2d_map(
         self._2d_last_updated = self._seq
         return obstacles, explored
 
+    def xy_to_grid_coords(self, xy: np.ndarray) -> Optional[np.ndarray]:
+        """convert xy point to grid coords"""
+        # TODO: tensor version of this
+        grid_xy = (xy / self.grid_resolution) + self.grid_origin[:2].cpu().numpy()
+        if np.any(grid_xy >= self.grid_size) or np.any(grid_xy < np.zeros(2)):
+            return None
+        else:
+            return xy
+
+    def grid_coords_to_xy(self, grid_coords: np.ndarray) -> np.ndarray:
+        """convert grid coordinate point to metric world xy point"""
+        assert grid_coords.shape[-1] == 2, "grid coords must be an Nx2 or 2d array"
+        # TODO: tensor version of this
+        return (grid_coords - self.grid_origin[:2].cpu().numpy()) * self.grid_resolution
+
+    def grid_coords_to_xyt(self, grid_coords: np.ndarray) -> np.ndarray:
+        """convert grid coordinate point to metric world xyt point"""
+        res = np.zeros(3)
+        res[:2] = self.grid_coords_to_xy(grid_coords)
+        return res
```

Comment:
These functions can all be written in torch

Reply:
fixed